python/pyspark/sql/pandas/conversion.py

```diff
@@ -360,12 +376,25 @@ def createDataFrame(  # type: ignore[misc]
 
         assert isinstance(self, SparkSession)
 
+        timezone = self._jconf.sessionLocalTimeZone()
+
+        if type(data).__name__ == "Table":
+            # `data` is a PyArrow Table
+            from pyspark.sql.pandas.utils import require_minimum_pyarrow_version
+
+            require_minimum_pyarrow_version()
```

Comment:
Would it be good to check pa.Table with isinstance at this point?

Reply:
Sure, I could add an assertion here just to be safe. I think the type hints in the function signature should prevent us from getting here if it's not a PyArrow Table but I'll add an assertion for good measure.

Reply:
Added in ee489f0.